### PR TITLE
Fix bugs with origin 1.2 rpm based upgrades.

### LINKED ISSUE #2167: Fail to upgrade Origin after updating openshift-ansible to the latest release
Hi. I have updated openshift-ansible today to the latest release. I tried to update Origin to v1.2.1 with the following command:

`# ansible-playbook -i /etc/ansible/hosts  openshift-ansible/playbooks/byo/openshift-cluster/upgrades/v3_2/upgrade.yml`

But it failed with this error:

```
TASK: [openshift_version | fail ] ********************************************* 
failed: [master-1.paas.srv.srvr.rbs.net] => {"failed": true}
msg: Detected openshift version 1.2.0 does not match requested openshift_release 3.2. 
You may need to adjust your yum repositories or specify an exact openshift_pkg_version. 
FATAL: all hosts have already failed -- aborting
```

I  have made this procedure many times before and never got this type of error.
Does anyone knows what this error can be?

Thanks in advance.

### PR DESCRIPTION
**Pull request by dgoodwin**
Fixes #2167

### CODE REVIEW

**Comment by sdodson** [comment]
aos-ci-test

**Comment by abutcher** [comment]
Retrying CI job.

File: playbooks/common/openshift-cluster/upgrades/v3_1_to_v3_2/pre.yml [hunk 1]
```diff
@@ -176,8 +176,12 @@
   - name: Verify OpenShift 3.2 RPMs are available for upgrade
     fail:
       msg: "OpenShift {{ avail_openshift_version.stdout }} is available, but 3.2 or greater is required"
-    when: not openshift.common.is_containerized | bool and not avail_openshift_version | skipped and avail_openshift_version.stdout | default('0.0', True) | version_compare('3.2', '<')
+    when: deployment_type != 'origin' and not openshift.common.is_containerized | bool and not avail_openshift_version | skipped and avail_openshift_version.stdout | default('0.0', True) | version_compare('3.2', '<')
```
Comment (sdodson) [inline playbooks/common/openshift-cluster/upgrades/v3_1_to_v3_2/pre.yml]: `version_compare(openshift_release,'<')` to cut down on the chances of updating one place to 3.3 in the future but not another?

Comment (dgoodwin) [inline playbooks/common/openshift-cluster/upgrades/v3_1_to_v3_2/pre.yml]: Updated. Will probably be able to improve this more soon when introducing 3.3 upgrade.

**Comment by sdodson** [comment]
LGTM

**Comment by sdodson** [comment]
aos-ci-test

**Comment by detiber** [comment]
:+1:

**Pull request merged**